google/cloud/spanner/internal/spanner_stub.cc

```diff
@@ -0,0 +1,257 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/spanner_stub.h"
+#include <google/spanner/v1/spanner.grpc.pb.h>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+namespace {
+
+namespace spanner_proto = ::google::spanner::v1;
+
+/**
+ * DefaultSpannerStub - a stub that calls Spanner's gRPC interface.
+ */
+class DefaultSpannerStub : public SpannerStub {
```

Comment:
While writing a unit test for this class is a pain, writing integration tests is needed. Can you at add a bug + TODO for the integration tests?